src/nvim/lua/vim.lua

```diff
@@ -439,6 +439,21 @@ function vim.notify(msg, log_level, opts) -- luacheck: no unused
   end
 end
 
+do
+  local warned = {}
+
+  --- Emit a warning message once per Nvim session.
+  ---
+  --- Warnings are emitted using |vim.notify()|.
+  ---
+  ---@param msg string Warning message to display to the user.
+  function vim.warn_once(msg)
```

Comment:
Since this is exposed to user space we should probably take log level as input too . That way plugins can use it for onetime error/info messages too . What do you think ?